packages/types/src/promotion/common/campaign.ts

```diff
@@ -0,0 +1,44 @@
+import { BaseFilterable } from "../../dal"
+import {
+  CampaignBudgetDTO,
+  CreateCampaignBudgetDTO,
+  UpdateCampaignBudgetDTO,
+} from "./campaign-budget"
+
+export interface CampaignDTO {
+  id: string
+  name?: string
+  description?: string
+  currency?: string
+  campaign_identifier?: string
+  starts_at?: Date
+  ends_at?: Date
+  campaign_budget?: CampaignBudgetDTO
+}
+
+export interface CreateCampaignDTO {
+  name: string
+  description?: string
+  currency?: string
+  campaign_identifier: string
+  starts_at: Date
+  ends_at: Date
+  campaign_budget?: CreateCampaignBudgetDTO
+}
+
+export interface UpdateCampaignDTO {
+  id: string
+  name?: string
+  description?: string
+  currency?: string
+  campaign_identifier?: string
+  starts_at?: Date
+  ends_at?: Date
+  campaign_budget?: Omit<UpdateCampaignBudgetDTO, "id">
+}
```

Comment:
suggestion: Move mutation DTOs to a file `mutations.ts` in the same folder. Think we should establish this as a convention.